### PR TITLE
add conf regex for build image

### PR DESCRIPTION
**Pull request by ktibi**

### CODE REVIEW

File: ansible/group_vars/all/kolla [hunk 1]
```diff
@@ -106,18 +106,28 @@ seed_container_image_sets:
 # List of regular expressions matching names of container images to build for
 # controllers.
 controller_container_image_regex_map:
+  - regex: aodh
+    enabled: "{{ kolla_enable_aodh | bool }}"
   - regex: barbican
     enabled: "{{ kolla_enable_barbican | bool }}"
+  - regex: ceilomter
```
Comment (markgoddard) [inline ansible/group_vars/all/kolla]: spelling

File: ansible/group_vars/all/kolla [hunk 2]
```diff
@@ -106,18 +106,28 @@ seed_container_image_sets:
...
   - regex: cinder
     enabled: True
   - regex: cron
     enabled: True
+  - regex: designate
+    enabled: "{{ kolla_enable_designate | bool }}"
   - regex: dnsmasq
     enabled: True
+  - regex: elasticsearch
+    enabled: "{{ kolla_enable_central_logging | bool }}"
```
Comment (markgoddard) [inline ansible/group_vars/all/kolla]: This should be kolla_enable_elasticsearch

File: ansible/group_vars/all/kolla [hunk 3]
```diff
@@ -132,6 +142,8 @@ controller_container_image_regex_map:
     enabled: True
   - regex: keystone
     enabled: True
+  - regex: kibana
+    enabled: "{{ kolla_enable_central_logging | bool }}"
```
Comment (markgoddard) [inline ansible/group_vars/all/kolla]: This should be kolla_enable_kibana

File: ansible/group_vars/all/kolla [hunk 4]
```diff
@@ -155,6 +167,8 @@ controller_container_image_regex_map:
     enabled: True
   - regex: sahara
     enabled: "{{ kolla_enable_sahara | bool }}"
+  - regex: skydive
+    enabled: "{{ kolla_enable_sahara | bool }}"
```
Comment (markgoddard) [inline ansible/group_vars/all/kolla]: s/sahara/skydive/

File: ansible/group_vars/all/kolla [hunk 5]
```diff
@@ -106,18 +106,28 @@ seed_container_image_sets:
 # List of regular expressions matching names of container images to build for
```
Comment (markgoddard) [inline ansible/group_vars/all/kolla:106]: We need defaults for all of these which match kolla's defaults.

kolla_enable_central_logging isn't used directly, but sets the defaults for elasticsearch and kibana (from kolla-ansible group_vars/all.yml):

```
enable_elasticsearch: "{{ 'yes' if enable_central_logging | bool or enable_freezer | bool or enable_osprofiler | bool or enable_skydive | bool else 'no' }}"
enable_kibana: "{{ 'yes' if enable_central_logging | bool else 'no' }}"
```

**Review by markgoddard — CHANGES_REQUESTED** [review]

File: ansible/group_vars/all/kolla (hunk 5 above)
Comment (ktibi) [reply ansible/group_vars/all/kolla:106]: done

File: ansible/group_vars/all/kolla (hunk 1 above)
Comment (ktibi) [reply ansible/group_vars/all/kolla]: done

File: ansible/group_vars/all/kolla (hunk 3 above)
Comment (ktibi) [reply ansible/group_vars/all/kolla]: done

File: ansible/group_vars/all/kolla (hunk 2 above)
Comment (ktibi) [reply ansible/group_vars/all/kolla]: done

File: ansible/group_vars/all/kolla (hunk 4 above)
Comment (ktibi) [reply ansible/group_vars/all/kolla]: done

File: ansible/group_vars/all/kolla (hunk 5 above)
Comment (markgoddard) [reply ansible/group_vars/all/kolla:106]: Sorry, I wasn't clear - we need to add the defaults with the other kolla_enable_* variables below here, then reference them in this list.

File: ansible/group_vars/all/kolla [hunk 6]
```diff
@@ -106,18 +106,28 @@ seed_container_image_sets:
...
   - regex: cinder
     enabled: True
   - regex: cron
     enabled: True
+  - regex: designate
+    enabled: "{{ kolla_enable_designate | bool }}"
   - regex: dnsmasq
     enabled: True
+  - regex: elasticsearch
+    enabled: "{{ 'yes' if enable_central_logging | bool or kolla_enable_elasticsearch | bool or enable_freezer | bool or enable_osprofiler | bool or enable_skydive | bool else 'no' }}"
```
Comment (markgoddard) [inline ansible/group_vars/all/kolla]: This should just be kolla_enable_elasticsearch, which should be set below to:

`kolla_enable_elasticsearch: {{ 'yes' if kolla_enable_central_logging | bool or kolla_enable_skydive | bool else 'no' }}`

File: ansible/group_vars/all/kolla [hunk 7]
```diff
@@ -132,6 +142,8 @@ controller_container_image_regex_map:
     enabled: True
   - regex: keystone
     enabled: True
+  - regex: kibana
+    enabled: "{{ 'yes' if enable_central_logging | bool or kolla_enable_kibana | bool else 'no' }}"
```
Comment (markgoddard) [inline ansible/group_vars/all/kolla]: As above for elasticsearch.

File: ansible/group_vars/all/kolla (hunk 5 above)
Comment (ktibi) [reply ansible/group_vars/all/kolla:106]: my bad, I do not know read

Comment (markgoddard) [reply ansible/group_vars/all/kolla:106]: The problem we have is that kayobe and kolla-ansible have separate variable namespaces. If you reference a variable in controller_container_image_regex_map, it needs to be defined somewhere. In your kayobe-config I suspect you have all of these variables defined, but in my kayobe-config I don't have kolla_enable_aodh or kolla_enable_ceilometer etc. defined. That's why we need to add default values for all of these like you have for kibana. The goal should be to be able to use the defaults.

You can test this like so:
```
kayobe configuration dump --var-name controller_container_image_regexes
{
    "localhost": [
        "cinder", 
        "cron", 
        "dnsmasq", 
        "fluentd", 
        "glance", 
        "haproxy", 
        "heat", 
        "horizon", 
        "iscsid", 
        "keepalived", 
        "keystone", 
        "kolla-toolbox", 
        "mariadb", 
        "memcached", 
        "neutron-server", 
        "neutron-\\(dhcp\\|l3\\|metadata\\|openvswitch\\)-agent", 
        "nova", 
        "openvswitch", 
        "rabbitmq", 
        "tgtd"
    ]
}
```

If I now make one of the variables undefined, (e.g. add `enabled: "{{ undefined_variable }}"` I get this:
```
kayobe configuration dump --var-name controller_container_image_regexes
{
    "localhost": "{{ controller_container_image_regex_map | selectattr('enabled') | map(attribute='regex') | list }}"
}
```

File: ansible/group_vars/all/kolla [hunk 8]
```diff
@@ -255,6 +269,8 @@ kolla_enable_magnum: "no"
 kolla_enable_murano: "no"
 kolla_enable_sahara: "no"
 kolla_enable_swift: "no"
+kolla_enable_elasticsearch: "{{ 'yes' if enable_central_logging | bool or enable_freezer | bool or enable_osprofiler | bool or enable_skydive | bool else 'no' }}"
```
Comment (markgoddard) [inline ansible/group_vars/all/kolla]: The variables in the `{{ }}` should all have `kolla_` prefixes. They should also all have a default defined.

File: ansible/group_vars/all/kolla [hunk 9]
```diff
@@ -255,6 +269,8 @@ kolla_enable_magnum: "no"
...
+kolla_enable_kibana: "{{ 'yes' if enable_central_logging | bool else 'no' }}"
```
Comment (markgoddard) [inline ansible/group_vars/all/kolla]: kolla_enable_central_logging

File: ansible/group_vars/all/kolla [hunk 10]
```diff
@@ -242,18 +256,24 @@ kolla_openstack_logging_debug: "False"
 ###############################################################################
 # Kolla feature flag configuration.
 
+kolla_enable_aodh: "no"
 kolla_enable_barbican: "no"
-kolla_enable_ceph: "no"
```
Comment (markgoddard) [inline ansible/group_vars/all/kolla]: Need ceph!

File: ansible/group_vars/all/kolla [hunk 11]
```diff
@@ -242,18 +256,24 @@ kolla_openstack_logging_debug: "False"
...
+kolla_enable_central_logging: "no"
+kolla_enable_ceilometer: "no"
+kolla_enable_designate: "no"
+kolla_enable_elasticsearch: "{{ 'yes' if kolla_enable_central_logging | bool or kolla_enable_freezer | bool or kolla_enable_osprofiler | bool or kolla_enable_skydive | bool else 'no' }}"
 kolla_enable_glance: "yes"
+kolla_enable_gnocchi: "no"
 kolla_enable_haproxy: "yes"
 kolla_enable_heat: "yes"
-kolla_enable_horizon: "yes"
```
Comment (markgoddard) [inline ansible/group_vars/all/kolla]: Need horizon!

File: ansible/group_vars/all/kolla [hunk 12]
```diff
@@ -242,18 +256,24 @@ kolla_openstack_logging_debug: "False"
 ###############################################################################
 # Kolla feature flag configuration.
 
+kolla_enable_aodh: "no"
 kolla_enable_barbican: "no"
-kolla_enable_ceph: "no"
+kolla_enable_central_logging: "no"
+kolla_enable_ceilometer: "no"
+kolla_enable_designate: "no"
+kolla_enable_elasticsearch: "{{ 'yes' if kolla_enable_central_logging | bool or kolla_enable_freezer | bool or kolla_enable_osprofiler | bool or kolla_enable_skydive | bool else 'no' }}"
```
Comment (markgoddard) [inline ansible/group_vars/all/kolla]: Freezer and osprofiler still undefined!

**Review by markgoddard — CHANGES_REQUESTED** [review]

**Review by markgoddard — APPROVED** [review]
Looks good, thanks

**Pull request merged**